### ISSUE TITLE
google_checks.xml incorrectly marks /** {@return the customer ID} */ as violation

### ISSUE THREAD

**Issue by PaulLieverse-TomTom** [checkstyle/checkstyle#18842]
Google Java Style Guide: https://google.github.io/styleguide/javaguide.html#s7.2-summary-fragment explicitly allows using `/** {@return the customer ID} */`.

<img width="1100" height="321" alt="Image" src="https://github.com/user-attachments/assets/bd092df6-b426-45cd-b06a-815aabde75e1" />

However, this now results in `SummaryJavadoc` reporting an error `Forbidden summary fragment.`

```bash
/var/tmp $ javac Test.java 

/var/tmp $ cat Test.java 
/** A test class. */
public class Test {

  /** {@return the customer ID} */
  static int customerId() {
    return 10;
  }
}

/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-13.1.0-all.jar -c google_checks.xml Test.java
Starting audit...
[WARN] /var/tmp/Test.java:4: Forbidden summary fragment. [SummaryJavadoc]
Audit done.
```

This seems to be due the regexp change in `google_checks.xml` done in https://github.com/checkstyle/checkstyle/pull/18241 to fix issue https://github.com/checkstyle/checkstyle/issues/17727 . Removing the final `|^[a-z]` from the regexp no longer triggers this violation (but may  re-introduce the issue fixed with that PR).

```bash
/var/tmp $ cat test_config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="SummaryJavadoc">
      <property name="id" value="patternFromGoogleChecks"/>
      <property name="forbiddenSummaryFragments"
                value="^@return the *|^This method returns |^A [{]@code [a-zA-Z0-9]+[}]( is a )|^[a-z]"/>
    </module>
    <module name="SummaryJavadoc">
      <property name="id" value="fixedPattern"/>
      <property name="forbiddenSummaryFragments"
                value="^@return the *|^This method returns |^A [{]@code [a-zA-Z0-9]+[}]( is a )"/>
    </module>
  </module>
</module>

/var/tmp $ java -Duser.language=en -Duser.country=US -jar checkstyle-13.1.0-all.jar -c test_config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:4: Forbidden summary fragment. [patternFromGoogleChecks]
Audit done.
Checkstyle ends with 1 errors.

```

**Comment by romani** [comment]
@guptapratykshh, can you help us fix this regression?

**Comment by Anushreebasics** [comment]
I am on it

**Comment by romani** [comment]
based on investigation at https://github.com/checkstyle/checkstyle/pull/20550#discussion_r3643703317

When user define javadoc:
`/** @return the customer ID */`

in html:
NO summary.
in details:
```
 Returns:
 the customer ID
```

When user define javadoc:
`/** {@return the customer ID} */`

in html:
summary is `Returns customer ID`
in details:
```
 Returns customer ID
 Returns:
 the customer ID
```


=========

now we violate  summary that start from lowercase in javadoc comment code.
real content in javadoc starts from lowercase, yes, BUT what user see in html is prepend with "Returns " word so what user see does not start with lowercase. 

**question is:**
If user defined bad pattern in checkstyle config what user is validating ? 
html rendering results (text) or technical content (html and inlinetags) ?

our doc does not make any statement on this.

**Answer:**
for regular not tagged summary, it does not put in validation html tags so validation happens on text after html rendering.
https://github.com/checkstyle/checkstyle/blob/aac71c9d1c7757df97ed492ec671d7dfafb9e22a/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java#L444

So for inline tags we should do same and prepend "Returns " before applying regexp.

**Comment by romani** [comment]
fix is merged
